plugins-builtin/src/test/scala/za/co/absa/enceladus/plugins/builtin/errorinfo/mq/ErrorInfoSenderPluginSuite.scala

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.plugins.builtin.errorinfo.mq
+
+import java.time.Instant
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+import com.typesafe.config.{ConfigFactory, ConfigValueFactory}
+import org.scalatest.BeforeAndAfterAll
+import za.co.absa.abris.avro.read.confluent.SchemaManager
+import org.scalatest.{FlatSpec, Matchers}
+import za.co.absa.enceladus.plugins.builtin.common.mq.kafka.KafkaConnectionParams
+import za.co.absa.enceladus.plugins.builtin.errorinfo.DceErrorInfo
+import za.co.absa.enceladus.plugins.builtin.errorinfo.mq.ErrorInfoSenderPluginSuite.{TestingErrCol, TestingRecord}
+import za.co.absa.enceladus.plugins.builtin.errorinfo.mq.kafka.KafkaErrorInfoPlugin
+import za.co.absa.enceladus.utils.testUtils.SparkTestBase
+import org.apache.spark.sql.DataFrame
+import za.co.absa.enceladus.plugins.builtin.errorinfo.params.ErrorInfoPluginParams
+import za.co.absa.enceladus.plugins.builtin.errorinfo.params.ErrorInfoPluginParams.ErrorSourceId
+
+class ErrorInfoSenderPluginSuite extends FlatSpec with SparkTestBase with Matchers with BeforeAndAfterAll {
```

Comment:
Testing of the plugin covers 
1. internal processing logic, 
2. confluent_avro serialization and 
3. API-parameter map conversion checking.

What it does not cover directly is the actual message sending to Kafka. Do we want this part to be covered, too?

Reply:
Probably no need to test the actual sending.

The tests are really good, IMHO.